src/plugins/discover/public/application/view_components/canvas/top_nav.tsx

```diff
@@ -35,10 +36,16 @@ export const TopNav = ({ opts }: TopNavProps) => {
     },
     data,
     chrome,
+    osdUrlStateStorage,
   } = services;
 
   const topNavLinks = savedSearch ? getTopNavLinks(services, inspectorAdapters, savedSearch) : [];
 
+  connectStorageToQueryState(services.data.query, osdUrlStateStorage, {
```

Comment:
Do you know why this works? is it removing the useEffect or moving the location?